json-rpc/src/stream_rpc/README.md

```diff
@@ -0,0 +1,85 @@
+# Streaming RPC
+The goal of the streaming RPC is to solve the following problems:
+- There is significant overhead to making a new connection for each request, [compared to WebSockets or Server Side Events](http://www.diva-portal.se/smash/get/diva2:1133465/FULLTEXT01.pdf)
+- Polling is wasteful vs server-side streaming, especially for sparse event streams
+
+Simplified flow:
+![Overall stream flow](../../docs/images/stream_rpc_overall_flow.png)
+
+This document assumes the perspective of a fullnode. Any relative terms (i.e `incoming`) are therefore intended to mean `incoming to the fullnode`.
+
+Configuration for the streaming RPC endpoint is in `/config/src/config/json_rpc_config.rs`.
+
+## Transports
+Transports handle connections to the outside world.
+Transports are responsible for mapping external representations of communication to internal representations, and vice versa.
+
+The currently supported transports are:
+- Websockets
+
+Planned transports are:
+- Server-Side Events
+
+For example: in the case of WebSockets, incoming messages have the form of `Result<warp::filters::ws::Message, warp::Error>`.
+Different transports will have different external message representations, and so the `ConnectionManager` expects a `dyn Stream`
```

Comment:
theoretically you could have made an enum containing a set of pre-defined streams since the types are fixed instead of dyn Stream... there's a good example of this in secure storage: https://github.com/diem/diem/blob/main/secure/storage/src/storage.rs

maybe in another PR if you want to shave of microseconds in overhead :)